src/controllers/dynakube/activegate/reconciler.go

```diff
@@ -70,15 +70,24 @@ func (r *Reconciler) Reconcile() error {
 		return err
 	}
 
-	var caps = capability.GenerateActiveGateCapabilities(r.dynakube)
+	caps := capability.GenerateActiveGateCapabilities(r.dynakube)
+
+	if r.dynakube.IsSyntheticMonitoringEnabled() {
+		for _, cap := range caps {
+			if cap.Enabled() && cap.ShortName() != capability.SyntheticName {
+				return errors.New("synthetic capability can't be enabled with other capabilities in the same DynaKube")
+			}
+		}
+	}
+
 	for _, agCapability := range caps {
 		if agCapability.Enabled() {
-			return r.createCapability(agCapability)
+			err = r.createCapability(agCapability)
```

Comment:
Plausibly the [said manifest](https://github.com/Dynatrace/dynatrace-operator/pull/1581#discussion_r1136601730) maps to two `StatefulSets`.

Reply:
Reverted to the original.